src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -3568,16 +3588,30 @@ def _sync_runs(self) -> None:
         from zenml.zen_stores.migrations.alembic import AlembicVersion
 
         try:
-            with Session(self.engine) as session:
-                # We use the alembic version table as a shared resource that we
-                # can lock to prevent multiple processes from syncing runs at
-                # the same time.
-                logger.debug("Syncing pipeline runs...")
-                session.query(AlembicVersion).with_for_update().all()
-                self._sync_runs_with_lock(session)
-                logger.debug("Pipeline runs sync complete")
+            # This is to synchronize the locally running threads so that only
+            # one thread attempts to sync the runs at any given time. The
+            # timeout is to ensure that we don't block the threads for too long.
+            if self.sync_lock.acquire(timeout=30):
+                with Session(self.engine) as session:
+                    logger.debug("Syncing pipeline runs...")
+                    if self.config.driver != SQLDatabaseDriver.SQLITE:
+                        # This is to synchronize all server processes trying to
+                        # sync the pipeline runs at the same time. We use the
+                        # alembic version table as a shared resource that we can
+                        # lock to prevent multiple processes from syncing runs
+                        # at the same time.
+                        session.query(AlembicVersion).with_for_update().all()
```

Comment:
Do we even still need this code now that we lock it using `threading.Lock`?

Reply:
As the comment says, this is needed you're using multiple replicas in your server (multiple zenml server instance clones connected to the same database).